src/main/scala/org/broadinstitute/dsde/firecloud/service/DataUseRestrictionSupport.scala

```diff
@@ -176,6 +225,60 @@ trait DataUseRestrictionSupport extends LazyLogging {
       allDurFieldNames.map(AttributeName.withLibraryNS)) ++ preferred
   }
 
+  def generateStructuredUseRestrictionAttribute(request: StructuredDataRequest, ontologyDAO: OntologyDAO): Map[String, JsValue] = {
+    // get DS diseases (map over array of ints)
+    val diseaseCodesArray = getDiseaseNames(request.diseaseUseOnly, ontologyDAO).map("DS:" + _)
+
+    // create map of correct restrictions
+    val consentMap = Map(
+      AttributeName.withDefaultNS(ConsentCodes.GRU) -> AttributeBoolean(request.generalResearchUse),
+      AttributeName.withDefaultNS(ConsentCodes.HMB) -> AttributeBoolean(request.healthMedicalBiomedicalUseOnly),
+      AttributeName.withDefaultNS(ConsentCodes.NCU) -> AttributeBoolean(request.commercialUseProhibited),
+      AttributeName.withDefaultNS(ConsentCodes.NPU) -> AttributeBoolean(request.forProfitUseProhibited),
+      AttributeName.withDefaultNS(ConsentCodes.NMDS) -> AttributeBoolean(request.methodsResearchProhibited),
+      AttributeName.withDefaultNS(ConsentCodes.NAGR) -> AttributeBoolean(request.aggregateLevelDataProhibited),
+      AttributeName.withDefaultNS(ConsentCodes.NCTRL) -> AttributeBoolean(request.controlsUseProhibited),
+      AttributeName.withDefaultNS(ConsentCodes.RSPD) -> AttributeBoolean(request.pediatricResearchOnly),
+      AttributeName.withDefaultNS(ConsentCodes.IRB) -> AttributeBoolean(request.IRB)) ++ getGenderCodeMap(request.genderUseOnly)
+
+    // convert to array of consent codes
+    val consentCodes = consentMap.filter(_._2.value).map(_._1.name).toArray ++ diseaseCodesArray
+
+    //formatWithPrefix(request.prefix, StructuredDataResponse(consentCodes, 1.0, consentMap ++ Map(ConsentCodes.DS -> AttributeValueList(request.diseaseUseOnly.map(AttributeNumber(_))))))
+    StructuredDataResponse(consentCodes, 1.0, request.prefix, consentMap ++ Map(AttributeName.withDefaultNS(ConsentCodes.DS) -> AttributeValueList(request.diseaseUseOnly.map(AttributeNumber(_))))).formatWithPrefix
+  }
+
+  private def getDiseaseNames(diseaseCodes: Array[Int], ontologyDAO: OntologyDAO): Array[String] = {
+    getDiseaseNames(diseaseCodes.map("http://purl.obolibrary.org/obo/DOID_" + _.toString), ontologyDAO)
```

Comment:
I'm concerned about this. For example, if we have something tagged with Pharynx Cancer, http://purl.obolibrary.org/obo/DOID_0060119, the `int` part of that DOID is `60119`, not `0060119` and http://purl.obolibrary.org/obo/DOID_60119 is not a valid DOID url.

Reply:
So we need to use `String` instead of `Int` to represent these things. I wonder if there's any reason to use just the `0060119` part or if we should always use the full `http://purl.obolibrary.org/obo/DOID_0060119`.

Reply:
One reason to use the whole URI is namespacing - it positively identifies which ontology you're referencing. This is similar to the use of URI namespaces in XML.

Besides, it's just darn convenient when debugging 😄